core/src/main/java/com/kumuluz/ee/factories/AgroalDataSourceFactory.java

```diff
@@ -0,0 +1,171 @@
+        } catch (SQLException e) {
+            throw new KumuluzServerException("Failed to create XaDataSource", e);
+        }
+    }
+
+    private static void setDatabaseCredentials(AgroalConnectionFactoryConfigurationSupplier connectionFactory, String username, String password) {
+        if (!StringUtils.isNullOrEmpty( username )) {
+            connectionFactory.principal(new NamePrincipal( username ));
+        }
+
+        if (!StringUtils.isNullOrEmpty( password )) {
+            connectionFactory.credential(new SimplePassword( password ));
+        }
+    }
+
+    private static void setConnectionPoolConfiguration(AgroalConnectionPoolConfigurationSupplier pool, AgroalConnectionFactoryConfigurationSupplier connectionFactoryConfig, DataSourcePoolConfig dscp) {
+        Optional.ofNullable( dscp.getAutoCommit() ).ifPresent(connectionFactoryConfig::autoCommit);
+
+        Optional.ofNullable( dscp.getConnectionTimeout() ).map(Duration::ofMillis).ifPresent(pool::acquisitionTimeout);
+
+        Optional.ofNullable( dscp.getIdleTimeout() ).map(Duration::ofMillis).ifPresent(pool::reapTimeout);
+
+        // Unknown or non-existent configuration in Agroal
+        Optional.ofNullable( dscp.getMaxLifetime() ).ifPresent((v) -> logMessageDepreciatedProperty("maxLifetime") );
+
+        Optional.ofNullable( dscp.getMaxSize() ).ifPresent(pool::maxSize);
+
+        if (!StringUtils.isNullOrEmpty( dscp.getName() )) {
+            // Unknown or non-existent configuration in Agroal
+            logMessageDepreciatedProperty("name");
+        }
+
+        // Unknown or non-existent configuration in Agroal
+        Optional.ofNullable( dscp.getInitializationFailTimeout() ).ifPresent((v) -> logMessageDepreciatedProperty("initializationFailTimeout") );
+
+        // Unknown or non-existent configuration in Agroal
+        Optional.ofNullable( dscp.getIsolateInternalQueries() ).ifPresent((v) -> logMessageDepreciatedProperty("isolateInternalQueries") );
+
+        // Unknown or non-existent configuration in Agroal
+        Optional.ofNullable( dscp.getAllowPoolSuspension() ).ifPresent((v) -> logMessageDepreciatedProperty("allowPoolSuspension") );
+
+        // Unknown or non-existent configuration in Agroal
+        Optional.ofNullable( dscp.getReadOnly() ).ifPresent((v) -> logMessageDepreciatedProperty("readOnly") );
+
+        // Unknown or non-existent configuration in Agroal
+        Optional.ofNullable( dscp.getRegisterMbeans() ).ifPresent((v) -> logMessageDepreciatedProperty("registerMbeans") );
+
+        Optional.ofNullable( dscp.getValidationTimeout() ).map(Duration::ofMillis).ifPresent(pool::validationTimeout);
+
+        Optional.ofNullable( dscp.getLeakDetectionThreshold() ).map(Duration::ofMillis).ifPresent(pool::leakTimeout);
+
+        // Unknown or non-existent configuration in Agroal
+        Optional.ofNullable( dscp.getMinIdle() ).ifPresent((v) -> logMessageDepreciatedProperty("minIdle") );
```

Comment:
it can be mapped to minSize no? And is it possible to add all others properties available in AgroalConnectionPoolConfigurationSupplier (initialSize for example)?

Reply:
Yes, this is possible. But for this first moment, I was concerned with supplying only the existing properties in Agroal and HikariCP. And for broken properties, show the message that it is depreciated.

Reply:
I understand but in a production environment it's important to be able to deeply configure the DS ; else performance tuning becomes quite impossible.

Reply:
Hi @salvadormarcos,

any updates on this, can we mark this implementation as complete?

Reply:
@gpor0 my point of view has not changed, I need to be able to fully configure the DS. If it's merged like this, I will have to fork the repo just to be able to specify few properties on the DS...

Reply:
Hi @salvadormarcos,
will you improve the implementation with missing properties, as suggested by @flavienlaurent? Do you have any other opened issues?

Reply:
Hi,

I understand @flavienlaurent's need to make detailed settings in DS. However, as commented in [issue 128](https://github.com/kumuluz/kumuluzee/issues/128#issuecomment-465087561), Agroal and Hikari would be kept side by side and Hikari would be used without the JTA component.

I can implement the missing properties, however Hikari does not have many existing properties in Agroal. The only problem would be when Hikari was the DS, several properties would simply be ignored. I believe that this can cause confusion for some users.

@gpor0 I can implement it if you agree with this situation. By the way, it's okay if I give attention to this next week? I am traveling and did not bring my laptop.

Reply:
Yes @salvadormarcos, please complete the implementation.

Reply:
Done! I analyzed the AgroalConnectionPoolConfigurationSupplier class and the only missing properties were: flushOnClose, initialSize, minSize and idleValidationTimeout.

Reply:
Great! We will make a review.